src/CompletionHandler.php

```diff
@@ -442,4 +435,19 @@ protected function getAllOptions()
             $this->application->getDefinition()->getOptions()
         );
     }
+
+    protected function getCommandNames()
+    {
+        $commands = array();
+
+        foreach ($this->application->all() as $command) {
```

Comment:
Please replace `as $command` with `as $name => $command`.

---

Optimization, see below comment for explanation.